src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,68 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.UserAgeException;
+import core.basesyntax.exception.UserLoginExistsException;
+import core.basesyntax.exception.UserLoginLengthException;
+import core.basesyntax.exception.UserPasswordLengthException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int USER_LOGIN_LENGTH_MIN = 6;
+    private static final int USER_PASSWORD_LENGTH_MIN = 6;
+    private static final int USER_AGE_MIN = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        String userLogin = user.getLogin();
+        String userPassword = user.getPassword();
+        int userAge = user.getAge();
+        if (isUserLoginExist(userLogin)) {
+            throw new UserLoginExistsException("User with this login: "
+                                                       + userLogin
+                                                       + " already exists");
+        } else if (!isUserLoginLengthOk(userLogin)) {
+            throw new UserLoginLengthException("User login length less than min length: "
+                                                       + USER_LOGIN_LENGTH_MIN
+                                                       + " try input another one");
+        } else if (!isUserPasswordLengthOk(userPassword)) {
+            throw new UserPasswordLengthException("User password length less than min length: "
+                                                          + USER_PASSWORD_LENGTH_MIN
+                                                          + " try input another one");
+        } else if (!isUserAgeOk(userAge)) {
+            throw new UserAgeException("User age less than min age: "
+                                               + USER_AGE_MIN);
+        }
+        storageDao.add(user);
+        return user;
```

Comment:
The method 'register' should validate if the passed User object is not null before attempting to access its properties. This is to avoid a potential NullPointerException.